### PR TITLE
Add op_id throughout op API

### PR DESCRIPTION
**Pull request by ry**
Ref https://github.com/denoland/deno/issues/2730

### CODE REVIEW

**Comment by piscisaureus** [comment]
I don't really understand what the purpose of moving the op_id out of the control buffer while leaving the other fields, (e.g. promise_id) in there. 
Would it not be better to shed the control buffer entirely then?

**Comment by afinch7** [comment]
Is this prep for dynamic op dispatching?

**Comment by ry** [comment]
@piscisaureus having the op_id put allows us to route to dispatcher - the control buffer contains all the parameters of the various ops - which is not something generally applicable to ops - like the filename parameter of Deno.stat shouldn’t be visible to Isolate::poll

**Comment by ry** [comment]
@afinch7 In part, yes. This allows each op to choose its own serialization mechanism. Something the read/write ops do currently through [a very terrible hack](https://github.com/denoland/deno/blob/43d099c0275878f6c6358eb0e9a0094d076d7c29/js/dispatch_minimal.ts#L21-L38):
https://github.com/denoland/deno/blob/43d099c0275878f6c6358eb0e9a0094d076d7c29/js/dispatch_minimal.ts#L6
In general this patch is a series of updates to the Op API outlined in https://github.com/denoland/deno/issues/2730

File: core/isolate.rs [hunk 1]
```diff
@@ -283,10 +287,12 @@ impl Isolate {
         // return value.
         // TODO(ry) check that if JSError thrown during respond(), that it will be
         // picked up.
-        let _ = isolate.respond(Some(&buf));
+        let op_id = 0; // Sync messages ignore op_id.
+        let _ = isolate.respond(Some((op_id, &buf)));
```
Comment (piscisaureus) [inline core/isolate.rs]: Maybe panic here if respond() errors?

Comment (ry) [reply core/isolate.rs]: done

**Pull request merged**